pandas/_libs/dtypes.pxd

```diff
@@ -0,0 +1,17 @@
+"""
+Common location for shared fused types
+"""
+
+from numpy cimport (
+    float32_t,
+    float64_t,
+    int64_t,
+    uint64_t,
+)
+
+ctypedef fused rank_t:
```

Comment:
can we give this a descriptive name, maybe `numeric_object_t` or similiar? (of course can update these later too which is fine)

Reply:
sounds like a good starting point, have updated